packages/amplify_api_plugin_interface/lib/src/GraphQL/GraphQLResponseError.dart

```diff
@@ -13,9 +13,56 @@
  * permissions and limitations under the License.
  */
 
-//TODO: This class needs to be fleshed out further to match the GraphQLError spec
+import '../util.dart';
+
+/// Contains an error produced via a GraphQL invocation. Corresponds to one
+/// entry in the `errors` field on a GraphQL response.
+///
+/// [locations] and [path] may be null.
 class GraphQLResponseError {
-  String message;
+  /// The description of the error.
+  final String message;
+
+  /// The locations of the error-causing fields in the request document.
+  final List<GraphQLResponseErrorLocation>? locations;
+
+  /// The key path of the error-causing field in the response's `data` object.
+  final List<dynamic>? path;
+
+  const GraphQLResponseError({
+    required this.message,
+    this.locations,
+    this.path,
+  });
+
+  Map<String, dynamic> toJson() => {
+        'message': message,
+        if (locations != null) 'locations': locations,
+        if (path != null) 'path': path,
```

Comment:
💯 
Wasn't aware of this syntax. Seems like we could/**should** be using it more throughout our code base

Reply:
For sure, yeah Dart has some cool little syntax items like this!